bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -168,6 +254,34 @@ bool ReplicaImp::validateMessage(MessageBase *msg) {
   }
 }
 
+/**
+ * asyncValidateMessage<T> This is a family of asynchronous message which just schedules
+ * the validation in a thread bag and returns peacefully. This will also translate the message
+ * into internal message.
+ *
+ * @param msg : This is the message whose validation is complex and should happen asynchronously.
+ *
+ * @return : returns nothing
+ */
+template <typename MSG>
+void ReplicaImp::asyncValidateMessage(MSG *msg) {
+  RequestThreadPool::getThreadPool().async(
```

Comment:
nit:
```
I don't think compiler do optimizations here. Slightly better:
```
static std::once_flag get_once;
static concord::util::ThreadPool tp;
std::call_once(get_once, [&]() { tp = RequestThreadPool::getThreadPool(); });
tp.async( ...

```

Reply:
Thanks @glevkovich , this is good. As the RequestThreadPool::getThreadPool() returns a reference to the static threadpool, we directly use the same. This is good for low level, and it definitely improves the understanding.

Reply:
@arc-vmware , yes - my suggestion was on how to improve the fetch of this reference (fetch only once and not millions of times), not on it's creation which is done only once.